### ISSUE TITLE
error of CUDNN_STATUS_ALLOC_FAILED / out of memory

### ISSUE THREAD

**Issue by yghlc** [jakeoung/u-net_pytorch#2]
Hi,
    I tried to train your model with the default batchSize (64)  on Quadro P5000 (16G), but it results in the error of out of memory.  I curious what is the GPU is needed your test?  Eventually, I have to reduce the batchSize to 8 for running this codes on Quadro P5000 (16G).  Thanks!

**Comment by jakeoung** [comment]
Oh, thank you for your interest on my incomplete codes^^ What dataset did you use? And you can check your memory by typing `nvidia-smi`.

**Comment by yghlc** [comment]
Thank you for your reply. The dataset is from high-resolution remote sensing images. I prepared the dataset myself. I set the batchsize as 8 and run on my GPU ( Quadro P5000). I also changed the size of each input image to 480 by 480.  The code occupied 9390MiB GPU memory, but it needs more than 12 GB at the beginning.

**Comment by jakeoung** [comment]
Maybe there are some issues related to cudnn. Could you try adding the following code in main.py?
`torch.backends.cudnn.enabled = False`

Or you can try setting `torch.backends.cudnn.benchmark` as False

**Comment by yghlc** [comment]
I added` torch.backends.cudnn.enabled = False` in the main.py. The situation was worse because the code occupied 13174 MiB GPU memory. Then I changed the `False` to `True`, the code occupied 9430MiB GPU memory, but it needed more than 12 GB at the beginning.

I removed` torch.backends.cudnn.enabled = False` and set` cudnn.benchmark` as `False`. Then the codes behave normally. The occupied GPU memory reached 8110MiB and became stable. 

I run the codes on Ubuntu 16.04, Python 2.7.12 and pytorch 0.1.12_2. Thank you for your help.
